src/context.rs

```diff
@@ -352,14 +352,15 @@ impl<'buf> Secp256k1<AllPreallocated<'buf>> {
     /// Creates a context from a raw context.
     ///
     /// # Safety
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
-    /// * The user must handle the freeing of the context(using the correct functions) by himself.
-    /// * Violating these may lead to Undefined Behavior.
     ///
+    /// This is highly unsafe due to the number of conditions that aren't checked, specifically:
+    ///
+    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer that was generated by *exactly*
```

Comment:
Pointer being generated is kinda vague. Maybe something like this:

> `raw_ctx` must be a valid pointer (live, aligned...) to the memory that was initialized by `secp256k1_context_preallocated_create` (either called directly, or from this library by one of the context creation methods - all of them call it internally)

One issue with this is this promises we always call that function somehow and there won't be alternative initialization methods that are incompatible with this. I think this is reasonable assumption, right @apoelstra ?

Reply:
Yep, that seems like a reasonable assumption for the indefinite future.

Reply:
Cool, thanks.